NeoML/src/Dnn/DnnBlob.cpp

```diff
@@ -218,13 +218,30 @@ CDnnBlob* CDnnBlob::GetCopy() const
 
 void CDnnBlob::CopyFrom(const CDnnBlob* other)
 {
-	NeoAssert(HasEqualDimensions(other));
-	switch(GetDataType()) {
+	NeoAssert( other != nullptr );
+	NeoAssert( GetDataType() == other->GetDataType() );
+	NeoAssert( HasEqualDimensions( other ) );
+	if( this == other ) {
+		return;
+	}
+	switch( GetDataType() ) {
 		case CT_Float:
-			mathEngine.VectorCopy( GetData<float>(), other->GetData<float>(), GetDataSize() );
+			if( &mathEngine == &other->GetMathEngine() ) {
+				mathEngine.VectorCopy( GetData<float>(), other->GetData<float>(), GetDataSize() );
+			} else {
+				float* buffer = const_cast<CDnnBlob*>( other )->GetBuffer<float>( 0, other->GetDataSize(), true );
```

Comment:
You has RAII-adapter `CDnnBlobBuffer`. Why don't use it?

Reply:
I forgot about it. Fixed.